File: Segundo-Bimestre/Parcial 1/Matrices/Matrices_python/Matrices3.py
# Inicializar las variables
n=0; i=0; j=0

# Ingresar el tamaño de la matriz
print("---EJERCICIO 4 - MATRICES---")
n = int(input("Ingrese el tamaño de la matriz: "))

# Declarar la matriz
A =  []

# Recorrer la matriz e ingresar los elementos de la matriz
print("Ingresar los elementos de la matriz")
for i in range (0, n):
    A.append([])
    for j in range (0, n):
        valor = int(input("A["+str(i)+"]["+str(j)+"] = "))
        A[i].append(valor)

# Presentar los elementos de la matriz
print("\nMATRIZ A ORIGINAL")
for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print(" ")

# Matriz con elementos pares reemplazados"
for i in range (0, n):
    for j in range (0, n):
        if (A[i][j] % 2 == 0):
            A[i][j]=0

print("\nMatriz A cambiando números pares por 0")
for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print(" ")

print("\nMatriz cuando i o j sea n-1")
for i in range (0, n):
    for j in range (0, n):
        if ((i==n-1) or (j==n-1)):
            A[i][j] = 1

for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print(" ")

print("\nLa matriz A según su posición es: ")
for i in range (0, n):
    for j in range (0, n):
        print("("+str(i)+")("+str(j)+")", end="   ")
    print(" ")

print("\nPresentar la matriz A cambiando números pares por 0 ")
for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print(" ")

File: Segundo-Bimestre/Parcial 1/Matrices/Matrices_java/src/matrices/Matrices3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package matrices;
import java.util.Scanner;
/**
 *
 * @author DELL
 */
public class Matrices3 {
    public static void main(String[] args) {
        Scanner leer = new Scanner(System.in);
        
        // Declarar e inicializar las variables
        int n, i, j;
        n = 0;
        
        // Ingresar el tamaño de la matriz
        System.out.println("---EJERCICIO 4-MATRICES---");
        System.out.print("Ingrese el tamaño de la matriz: ");
        n=leer.nextInt();
        
        // Declarar la matriz
        
        int [][] A= new int[n][n];
        
        // Recorrer la matriz e ingresar los elementos de la matriz
        System.out.println("\nIngresar los elementos de la matriz");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print("A ["+i+"] ["+j+"] = ");
                A[i][j] = leer.nextInt();
            }
        }
        
        // Presentar los elementos de la matriz
        System.out.println("Matriz A Original");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }
        
        System.out.println("\nMatriz con elementos pares reemplazados");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                if (A[i][j] % 2 == 0) {
                    A[i][j]=0;
                }
            }
        }

        System.out.println("Presentar la matriz A cambiando números pares por 0");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }

        System.out.println("\nMatriz cuando i o j sea n-1");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                if ((i==n-1)||(j==n-1)) {
                    A[i][j]=1;
                }
            }
        }
        
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }        
        
        System.out.println("\nLa matriz A según su posición es: ");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print("("+i+")("+j+")"+" ");
            }
            System.out.println("");
        }        

        System.out.println("Presentar la matriz A cambiando números pares por 0");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }        
    }
}
